app/models/discount.rb

```diff
@@ -0,0 +1,13 @@
+class Discount < ApplicationRecord
+  has_many :discount_items, dependent: :destroy
+  has_many :items, through: :discount_items
+  belongs_to :merchant
+
+  validates_presence_of :percent_discount
+  validates_presence_of :minimum_value, presence: true, unless: :minimum_quantity
+  validates_presence_of :minimum_quantity, presence: true, unless: :minimum_value
```

Comment:
This makes it so that one of the two of these needs to be provided in order to create a new discount. Nifty!